### PR TITLE
[SWDEV-303880] relocate and add MIOpen Porting Guide

### PR DESCRIPTION
**Pull request by bghimireamd**
The [link on how to port from cuDNN to MIOpen ](https://rocmdocs.amd.com/en/latest/Deep_learning/Deep-learning.html#porting-from-cudnn-to-miopen)was redirecting to [someone's private github](https://github.com/dagamayank/ROCm.github.io/blob/master/doc/miopen_porting_guide.pdf).

In this commit we convert the pdf version of MIOpen porting guide to markdown format. 

I have placed this file in doc/src/MIOpen_Porting_Guide.md

Rendered version can be viewed [here](https://github.com/ROCmSoftwarePlatform/MIOpen/blob/SWDEV-303880/doc/src/MIOpen_Porting_Guide.md). 

Rendered version of new [Readme](https://github.com/ROCmSoftwarePlatform/MIOpen/tree/SWDEV-303880#porting-from-cudnn-to-miopen)

### CODE REVIEW

**Review by JehandadKhan — CHANGES_REQUESTED** [review]
Can you please update the old link to the document and point it to this internal document. Please also attach a rendered version of the document so that it can be reviewed.

**Comment by junliume** [comment]
@bghimireamd could you follow up on the PR?

**Review by junliume — APPROVED** [review]

File: doc/src/MIOpen_Porting_Guide.md [hunk 1]
```diff
@@ -0,0 +1,2041 @@
...
+
+## Helpful MIOpen Environment Variables
+`MIOPEN_ENABLE_LOGGING=1` – log all the MIOpen APIs called including the parameters passed to
+those APIs. \
+`MIOPEN_DEBUG_AMD_ROCM_PRECOMPILED_BINARIES=0` – disable Winograd convolution
+algorithm. \
+`MIOPEN_DEBUG_GCN_ASM_KERNELS=0` – disable hand-tuned asm. kernels for Direct convolution
+algorithm. Fall-back to kernels written in high-level language. \
+`MIOPEN_DEBUG_CONV_FFT=0` – disable FFT convolution algorithm. \
+`MIOPEN_DEBUG_CONV_DIRECT=0` – disable Direct convolution algorithm.
```
Comment (atamazov) [inline doc/src/MIOpen_Porting_Guide.md:28]: The _Helpful MIOpen Environment Variables_ section from doc/src/MIOpen_Porting_Guide.md must be revised.

- duplicates info from [DebugAndLogging.md](https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/doc/src
- outdated (e.g. description of MIOPEN_DEBUG_AMD_ROCM_PRECOMPILED_BINARIES is incorrect)
- incomplete

**Review by atamazov — COMMENTED** [review]
outdated/duplicated info

File: doc/src/MIOpen_Porting_Guide.md (hunk 1 above)
Comment (bghimireamd) [reply doc/src/MIOpen_Porting_Guide.md:28]: @atamazov I will create a separate PR for this.

**Pull request merged**